lib/served/resource/requestable.rb

```diff
@@ -0,0 +1,187 @@
+module Served
+  module Resource
+    module Requestable
+      extend ActiveSupport::Concern
+
+      # raised when a handler is defined if the method doesn't exist or if a proc isn't supplied
+      class HandlerRequired < StandardError
+
+        def initialize
+          super 'a handler is required, it must be a proc or a valid method'
+        end
+
+      end
+
+      HEADERS = { 'Content-type' => 'application/json', 'Accept' => 'application/json' }
+
+      included do
+        include Configurable
+
+        class_configurable :resource_name do
+          name.split('::').last.tableize
+        end
+
+        class_configurable :host do
+          Served.config[:hosts][parent.name.underscore.split('/')[-1]] || Served.config[:hosts][:default]
+        end
+
+        class_configurable :timeout do
+          Served.config.timeout
+        end
+
+        class_configurable :_headers do
+          HEADERS
+        end
+
+        class_configurable :handlers, default: {}
+
+        class_configurable :template do
+          '{/resource*}{/id}.json{?query*}'
+        end
+
+        handle((200..201), :load)
+        handle([204, 202]) { attributes }
+
+        # 400 level errors
+        handle(301) { Resource::MovedPermanently }
+        handle(400) { Resource::BadRequest }
+        handle(401) { Resource::Unauthorized }
+        handle(403) { Resource::Forbidden }
+        handle(404) { Resource::NotFound }
+        handle(405) { Resource::MethodNotAllowed }
+        handle(406) { Resource::NotAcceptable }
+        handle(408) { Resource::RequestTimeout }
+        handle(422) { Resource::UnprocessableEntity }
+
+        # 500 level errors
+        handle(500) { Resource::InternalServerError }
+        handle(503) { Resource::BadGateway }
+
```

Comment:
as with the exceptions. I'd try to do this in a general way to handle all possible codes to not restrict ourselves.